### PR TITLE
Cleanups, the IntelliJ way

### PR DESCRIPTION
**Pull request by magicus**
I ran all the IntelliJ analyzers, and fixed the places where I agreed that the code was sub-optimal (about 10%, so this is not very efficient). Also found and fixed a couple of real bugs.

### CODE REVIEW

File: common/src/main/java/com/wynntils/mc/utils/objects/ServerIcon.java [hunk 1]
```diff
@@ -101,24 +101,25 @@ private synchronized void loadServerIcon(ResourceLocation destination) {
             return;
         }
 
-        NativeImage nativeImage;
         try {
-            nativeImage = NativeImage.fromBase64(iconString);
+            try (NativeImage nativeImage = NativeImage.fromBase64(iconString)) {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/mc/utils/objects/ServerIcon.java:105]: Not sure about this but it should be fine because it is disabled.

**Review by kristofbolyai — COMMENTED** [review]
Are you sure the final changes don't conflict with reflection/gson? If yes this is fine.

**Comment by magicus** [comment]
I'm not sure how I verify if the gson stuff breaks anything. I accepted a lot of these changes at first, but I actually thought I reverted them in the end, since I was not sure. Did I miss any?

File: common/src/main/java/com/wynntils/mc/utils/objects/ServerIcon.java (hunk 1 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/mc/utils/objects/ServerIcon.java:105]: I looked at the NativeImage code and it really looked like it should have it's close() method called. So I agreed with IntelliJ and said that this was an autocloseable miss.

**Comment by magicus** [comment]
I removed the finals from IdentificationProfile. Could not find any more for gson classes.

**Pull request merged**